pkg/endpoint/bpf.go

```diff
@@ -543,12 +543,12 @@ func (e *Endpoint) regenerateBPF(regenContext *regenerationContext) (revnum uint
 	}
 
 	// No need to compile BPF in dry mode.
-	if option.Config.DryMode {
+	if e.isMetadata(MetadataFakeEndpoint) {
 		return e.nextPolicyRevision, false, nil
 	}
 
 	// Skip BPF if the endpoint has no policy map
-	if !e.HasBPFPolicyMap() {
+	if e.isMetadata(MetadataSkipBPFPolicy) {
```

Comment:
Same comment as above (L 205), but for `HasBPFPolicyMap`.

Reply:
I've marked `HasBPFPolicyMap` as deprecated.